vent/sqlsol/parser.go

```diff
@@ -19,21 +22,63 @@ type Parser struct {
 	AbiSpec   *abi.AbiSpec
 }
 
-// NewParser receives a sqlsol event specification file stream
-// and returns a pointer to a filled parser structure
-// that contains event types mapped to SQL column types
-// and Event tables structures with table and columns info
-func NewParser(byteValue []byte) (*Parser, error) {
-
-	tables := make(types.EventTables)
+// NewParserFromBytes creates a Parser from a stream of bytes
+func NewParserFromBytes(bytes []byte) (*Parser, error) {
 	eventSpec := types.EventSpec{}
 
-	// parses json specification stream
-	if err := json.Unmarshal(byteValue, &eventSpec); err != nil {
+	if err := json.Unmarshal(bytes, &eventSpec); err != nil {
 		return nil, errors.Wrap(err, "Error unmarshalling eventSpec")
 	}
 
+	return NewParserFromEventSpec(eventSpec)
+}
+
+// NewParserFromFile creates a Parser from a file
+func NewParserFromFile(file string) (*Parser, error) {
+	bytes, err := readFile(file)
+	if err != nil {
+		return nil, errors.Wrap(err, "Error reading eventSpec file")
+	}
+
+	return NewParserFromBytes(bytes)
+}
+
+// NewParserFromFolder creates a Parser from a folder containing spec files
```

Comment:
As there is no use for `info` parameter in `filepath.Walk`, that function call could be clearer written `filepath.Walk(folder, func(path string, _ os.FileInfo, err error)`

Reply:
:+1: